gravitee-am-repository/gravitee-am-repository-mongodb/src/main/java/io/gravitee/am/repository/mongodb/management/MongoRateLimitRepository.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.repository.mongodb.management;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.reactivestreams.client.MongoCollection;
+import io.gravitee.am.common.utils.RandomString;
+import io.gravitee.am.model.RateLimit;
+import io.gravitee.am.repository.management.api.RateLimitRepository;
+import io.gravitee.am.repository.management.api.search.RateLimitCriteria;
+import io.gravitee.am.repository.mongodb.management.internal.model.RateLimitMongo;
+import io.reactivex.Completable;
+import io.reactivex.Maybe;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import org.bson.conversions.Bson;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.mongodb.client.model.Filters.and;
+import static com.mongodb.client.model.Filters.eq;
+
+/**
+ * @author Ashraful Hasan (ashraful.hasan at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Component
+public class MongoRateLimitRepository extends AbstractManagementMongoRepository implements RateLimitRepository {
+    private static final String RATE_LIMIT = "rate_limit";
+    private static final String FIELD_FACTOR_ID = "factorId";
+
+    private MongoCollection<RateLimitMongo> rateLimitCollection;
+
+    @PostConstruct
+    public void init() {
+        rateLimitCollection = mongoOperations.getCollection(RATE_LIMIT, RateLimitMongo.class);
+        super.init(rateLimitCollection);
```

Comment:
Index creation is missing.
Please name explicitly the indexes with a name less than 64 characters as there a limit on DocumentDB as far as I know.

Reply:
index added